lib/Onnxifi/Base.cpp

```diff
@@ -157,7 +164,48 @@ onnxStatus Graph::setIOAndRun(uint32_t inputsCount,
     ctx->getPlaceholderBindings()->insert(outPhPtr, std::move(t));
   }
 
-  return run(std::move(ctx), outputEvent, std::move(phNameToOnnxTensorOutputs));
+  return run(std::move(ctx), outputEvent, std::move(phNameToOnnxTensorOutputs),
+             traceEvents);
+}
+
+void Graph::setTraceEvents(onnxTraceEventList *traceEvents,
+                           const TraceContext &traceContext) {
+  if (!traceEvents) {
+    return;
+  }
+
+  std::vector<onnxTraceEvent *> traceEventsVec;
+  for (const auto &glowTraceEvent : traceContext.getTraceEvents()) {
+    auto *traceEvent = new onnxTraceEvent();
+    assert(
+        glowTraceEvent.type.size() == 1 &&
+        "Events with types longer than a single char not supported by onnxifi");
+    traceEvent->eventType = glowTraceEvent.type[0];
+    traceEvent->timestamp = glowTraceEvent.timestamp;
+    traceEvent->tid = glowTraceEvent.tid;
```

Comment:
It would be good to include `args` in the onnxTraceEvent since thats were we'll put per run metadata. Not sure if you can do maps

Reply:
Hmm yeah ok that will require changing the interface so let's do it in a followup